broker/src/main/java/io/zeebe/broker/system/partitions/ZeebePartition.java

```diff
@@ -518,7 +573,15 @@ private void stepByStepClosing(
   @Override
   public <T extends RaftLogEntry> void onCommit(final Indexed<T> indexed) {
     if (indexed.type() == ZeebeEntry.class) {
-      this.logStream.setCommitPosition(indexed.<ZeebeEntry>cast().entry().highestPosition());
+      actor.run(
+          () -> {
+            final long commitPosition = indexed.<ZeebeEntry>cast().entry().highestPosition();
```

Comment:
I dont think that this is necessary. Because at some point we will get the next commit position and we also seek to the last committed position, so for me there is no need for that?

Reply:
There is still a chance for race condition, right? logstream seek to last event, onCommit listner was invoke but ZeebePartition.logstream is null as the start is not completed. Then the commit position is one event behind.  commit position gets updated only if there are more commits. As said in a comment on the other draft PR, this is not a problem on the leader (now) because it doesn't start committing until stream processor is up. But it is possible on followers.

Reply:
So you expect that we never commit again ? :sweat_smile:

Reply:
:D What if there are no more requests?

Reply:
I did it like this for it to be more safe. But it might not be necessary now as followers never need to know commit position.

Reply:
I think it is a good idea, but I'm still unsure whether it adds to much complexity which is not really needed. :man_shrugging:

Reply:
Btw does it mean that we collect all commits on follower? didn't we go out of mem at some point?

Reply:
No. Once the logstream is opened, notifications will be cleared.